core/chains/evm/headtracker/head_listener.go

```diff
@@ -28,32 +32,62 @@ var (
 	}, []string{"evmChainID"})
 )
 
-type headListener struct {
-	config           Config
-	ethClient        evmclient.Client
+type headListener[
+	H commontypes.Head[BLOCK_HASH],
+	HTH htrktypes.Head[H, BLOCK_HASH, ID],
+	S commontypes.Subscription,
+	ID txmgrtypes.ID,
+	BLOCK_HASH commontypes.Hashable,
+	CLIENT htrktypes.Client[HTH, S, ID, BLOCK_HASH],
```

Comment:
isn't this an extraneous generic type? you already have all the subcomponents as generic types so the definition could be moved into the struct

Reply:
that makes sense, thanks for the explanation over slack! We dont need the generic type declaration